server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -538,7 +538,7 @@ public Iterator<Setting<?>> settings() {
     static final String KEY_MAPPINGS_HASH = "mappings_hash";
     static final String KEY_ALIASES = "aliases";
     static final String KEY_ROLLOVER_INFOS = "rollover_info";
-    static final String KEY_MAPPINGS_UPDATED_VERSION = "mappings_updated_version";
+    static final String KEY_INDEX_VERSION_WATERMARK = "index_version_watermark";
```

Comment:
I don't **think** there's any BwC requirements around the name for json output...?

Reply:
This seems to be a field name in json, I think it is a subject to BwC as there are both json writing and parsing.

Reply:
Does the compatibility need to go both ways, or can I just read the old `mappings_updated_version` and put it in the new field?

Reply:
>Does the compatibility need to go both ways,

Do we need to remove `mappings_updated_version`?

I wonder if it is safer to do it as adding a new field?
It could be initialized with the same value as `mappings_updated_version` the first time.